ispc: add x86_64-darwin to platforms.

athas opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

athas committed:
ispc: add x86_64-darwin to platforms.

marsam commented:
@GrahamcOfBorg eval
@GrahamcOfBorg build ispc

c0bw3b commented:
@GrahamcOfBorg build ispc

c0bw3b commented:
On Darwin:
```
1 / 1360 tests FAILED compilation:
        ./tests/transcendentals-10-0.ispc
NEW COMPFAILS:
        ./tests/transcendentals-10-0.ispc
```

athas commented:
Addressed in #71134 in the only robust way.

c0bw3b commented:
Recent history on Hydra does not show any failures for amd64-linux.
Can't we just set `doCheck = !stdenv.isDarwin;` as part of this PR instead ? (julia or nim are similar compilers examples)

athas commented:
Sure, but what happens next time this flaky test suite fails?  It's already been reduced to pointlessness, since to work reliably on Hydra, it only uses SSE2 instructions, and *nobody* interested in using ISPC is going to stick to SSE2.

marsam committed:
ispc: only check on Linux

The test suite tries to execute `transcendentals` tests, which is not
expected to work anywhere except Linux.
